### PR TITLE
docs(dotnet): examples for dialogs, fixes

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/dialogs.md [hunk 1]
```diff
@@ -110,3 +120,12 @@ def handle_dialog(dialog):
 page.on('dialog', lambda: handle_dialog)
 page.close(run_before_unload=True)
 ```
+
+```csharp
+page.Dialog += (_, dialog) =>
+{
+    Assert.Equal("beforeunload", dialog.Type);
```
Comment (kblok) [inline docs/src/dialogs.md:127]: Do we need to show this line?

**Review by kblok — APPROVED** [review]

File: docs/src/api/class-frame.md [hunk 2]
```diff
@@ -1424,8 +1424,8 @@ with frame.expect_navigation():
 ```
 
 ```csharp
-Task.WaitAll(frame.WaitForNavigationAsync(),
-            frame.ClickAsync("a.delayed-navigation")); // clicking the link will indirectly cause a navigation
+Task.WhenAll(frame.WaitForNavigationAsync(),
+             frame.ClickAsync("a.delayed-navigation")); // clicking the link will indirectly cause a navigation
```
Comment (kblok) [inline docs/src/api/class-frame.md:1428]:
```suggestion
Task.WhenAll(
    frame.WaitForNavigationAsync(),
    frame.ClickAsync("a.delayed-navigation")); // clicking the link will indirectly cause a navigation
```

**Review by kblok — APPROVED** [review]

File: docs/src/input.md [hunk 3]
```diff
@@ -751,11 +751,8 @@ file_chooser.set_files("myfile.pdf")
 
 ```csharp
 var waitForChooserTask = page.WaitForFileChooserAsync();
-await Task.WhenAll(
-    waitForChooserTask,
-    page.ClickAsync("upload")
-);
-await waitForChooserTask.Result.SetFilesAsync("myfile.pdf");
+await page.ClickAsync("upload");
+(await waitForChooserTask).SetFilesAsync("myfile.pdf");
```
Comment (dgozman) [inline docs/src/input.md:755]: Missing await for SetFilesAsync

**Review by dgozman — APPROVED** [review]

File: docs/src/dialogs.md (hunk 1 above)
Comment (pavelfeldman) [reply docs/src/dialogs.md:127]: dunno!

**Pull request merged**